From 3e34c067eab9f36be72cad558ad0d9a63c8596d9 Mon Sep 17 00:00:00 2001
From: Uri Eliyahu <uriel@marvell.com>
Date: Wed, 5 Feb 2014 18:56:40 +0200
Subject: [PATCH 1340/1825] fix: pp2: big endian debug command bug fix

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit dcc974969bb760fb7efe02378694e803a9b71379

	- debug flags command use inccorect set_bit/clear_bit functions

Change-Id: I56b85a91415452b63222fc33a6907adf76b8eb15
Signed-off-by: Uri Eliyahu <uriel@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/5467
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Dmitri Epshtein <dima@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 .../mv_drivers_lsp/mv_pp2/net_dev/mv_netdev.c      |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/plat-armada/mv_drivers_lsp/mv_pp2/net_dev/mv_netdev.c b/arch/arm/plat-armada/mv_drivers_lsp/mv_pp2/net_dev/mv_netdev.c
index cc94534..fe46e2b 100644
--- a/arch/arm/plat-armada/mv_drivers_lsp/mv_pp2/net_dev/mv_netdev.c
+++ b/arch/arm/plat-armada/mv_drivers_lsp/mv_pp2/net_dev/mv_netdev.c
@@ -476,9 +476,9 @@ int mv_eth_ctrl_dbg_flag(int port, u32 flag, u32 val)
 		return -ENODEV;
 
 	if (val)
-		set_bit(bit_flag, (unsigned long *)&(pp->dbg_flags));
+		pp->dbg_flags |= (1 << bit_flag);
 	else
-		clear_bit(bit_flag, (unsigned long *)&(pp->dbg_flags));
+		pp->dbg_flags &= ~(1 << bit_flag);
 #endif /* CONFIG_MV_ETH_DEBUG_CODE */
 
 	return 0;
-- 
1.7.5.4

